src/withStyles.jsx

```diff
@@ -228,20 +231,27 @@ export function withStyles(
       }
     }
 
+    const ForwardedWithStyles = React.forwardRef((props, ref) => (
```

Comment:
I think forwardRef was added in 16.3, but react-with-styles currently works all the way down to 0.14 https://github.com/airbnb/react-with-styles/blob/59a9548c7d18d8bf8bcf8a99cea85548114edf73/package.json#L81

Do you think it is possible to do this in a backwards compatible way, e.g. by detecting the presence of `React.forwardRef`

Reply:
The only thing I can think of would be adding an alternative entry point, and leave this one as-is - forwardRef doesn't really have any alternative in React < 16.3 that I'm aware of.

Reply:
Would it work to do something like this? (along with a guard in the render function so it conditionally passes it down to the wrapped component).

```js
const ForwardedWithStyles = typeof React.forwardRef === 'undefined'
  ? WithStyles
  : <WithStyles {...props} forwardedRef={ref} />;
```

That way, older versions continue to work as they do today, and 16.3+ will have the benefits of forwardRef.

Reply:
The `forwardedRef` propType (which i don't see added anywhere yet) would need to be conditionally applied, but that might work - altho it seems a bit weird to me to have an entire capability only be conditionally present based on react version.

Reply:
Yeah that's fair. Might as well make a breaking change. 16.3 was released almost 2 years ago now.

Reply:
> The forwardedRef propType (which i don't see added anywhere yet) would need to be conditionally applied

Can you explain what do you mean you don't see it added anywhere yet?

Reply:
An explicit propType that declares `forwardedRef` on `WithStyles`

Reply:
> An explicit propType that declares forwardedRef on WithStyles

Should this exist though? It should only be added by implicitly by `React.forwardRef`. A consumer of `withStyles` would not need to know about this prop. I don't think I'm understanding the requested change.

Reply:
Are we looking to have [something like this](https://stackoverflow.com/a/51127130)?

```jsx
const withStylesPropTypes = {
  refProp: PropTypes.oneOfType([
      PropTypes.func, 
      PropTypes.shape({ current: PropTypes.any })
  ])
}
```

Reply:
You could use `ref` from `airbnb-prop-types` :-p

Reply:
Nice. I added this change

```jsx
```

Reply:
@ljharb @lencioni what're our current thoughts here? Should we [conditionally fork][1] the existing entry point, or create a [brand new][2] entry point? Or create a breaking change given the release age of 16.3 now?

[1]: https://github.com/airbnb/react-with-styles/pull/240#discussion_r382745271
[2]: https://github.com/airbnb/react-with-styles/pull/240#discussion_r382740336

Reply:
I think it makes sense to publish a version with conditionally forking, and then immediately follow up with a breaking change to remove the conditional fork.

Reply:
Makes sense to me.